mesonbuild/dependencies/dev.py

```diff
@@ -505,6 +506,59 @@ def get_methods():
         return [DependencyMethods.SYSTEM]
 
 
+class JDKSystemDependency(ExternalDependency):
+    def __init__(self, environment: 'Environment', kwargs: T.Dict[str, T.Any]):
+        super().__init__('jdk', environment, kwargs)
+
+        m = self.env.machines[self.for_machine]
+
+        java_home = os.environ.get('JAVA_HOME')
```

Comment:
I'd really like to move the JAVA_HOME detection into `envconfig.Properties` and `environment.Environment`, I just got all our environment variable handling centralized there. `BOOST_ROOT` is handled in a smiliar fashion.

This would also be handy if we were to change the `javac` detection to `$JAVA_HOME/bin/javac`, since you'd avoid doing multiple lookups.

Reply:
Are you referring to `_set_default_properties_from_env`?

Reply:
I think I am following you. HAving trouble getting a reference to Environment from detect_java_compiler() though.

Reply:
Nevermind I am dumb again